qt/widgets/instrumentview/src/InstrumentWidgetMaskTab.cpp

```diff
@@ -625,9 +625,9 @@ void InstrumentWidgetMaskTab::shapeChanged() {
       false; // this prevents resetting shape properties by doubleChanged(...)
   RectF rect = m_instrWidget->getSurface()->getCurrentBoundingRect();
   m_doubleManager->setValue(m_left, rect.x0());
-  m_doubleManager->setValue(m_top, rect.y1());
+  m_doubleManager->setValue(m_top, rect.y0());
   m_doubleManager->setValue(m_right, rect.x1());
-  m_doubleManager->setValue(m_bottom, rect.y0());
+  m_doubleManager->setValue(m_bottom, rect.y1());
```

Comment:
This resolves the problem, but there is another issue that can be resolved here too. Currently the "top", "bottom", "left" and "right" labels are only correct if I draw a square starting with the bottom left corner and drag up to the right. Drawing a square in any other direction results in either the "top" and "bottom" or "left" and "right" being switchted. The same is true if I move one of the corners through the shape so that the left side becomes the right, for example. The following suggestion fixes that:
```
  m_doubleManager->setValue(m_left, std::min(rect.x0(), rect.x1()));
  m_doubleManager->setValue(m_top, std::max(rect.y0(), rect.y1()));
  m_doubleManager->setValue(m_right, std::max(rect.x0(), rect.x1()));
  m_doubleManager->setValue(m_bottom, std::min(rect.y0(), rect.y1()));
```

But this only works properly if the code around line 735 is changed from this:
```
QRectF rect(
        QPointF(m_doubleManager->value(m_left), m_doubleManager->value(m_top)),
        QPointF(m_doubleManager->value(m_right),
m_doubleManager->value(m_bottom)));
```
to this:
```
    double x0 = std::min(m_doubleManager->value(m_left),
                         m_doubleManager->value(m_right));
    double x1 = std::max(m_doubleManager->value(m_left),
                         m_doubleManager->value(m_right));
    double y0 = std::min(m_doubleManager->value(m_top),
                         m_doubleManager->value(m_bottom));
    double y1 = std::max(m_doubleManager->value(m_top),
                         m_doubleManager->value(m_bottom));
    QRectF rect(QPointF(x0, y0), QPointF(x1, y1));
```
Then the values of top, bottom, left and right always reflect the rectangle on screen. Give it a go and you'll see what I mean.

Reply:
I see what you mean, and it would be clearer for the user indeed, but with this code, if the user set the field value of say ``left`` to a greater value than ``right``, then ``right`` and ``left`` will be at the same new value. Do you think this is the most intuitive way, or should the values be swapped instead in this case ?

Reply:
Hmm. If you type a value of `left` greater than `right` and press the Enter key, then it sets them as the same value. I didn't test this before.

If you enter the value and then click somehwere else (and don't hit Enter), then it does as you suggest, it swaps them over - which I think is a good solution.